### PR TITLE
Check for duplicate claims from the same user

### PR DESCRIPTION
**Pull request by kaapstorm**
:tropical_fish:, `&w=1`

@proteusvacuum @phillipm cc @nickpell @snopoke

### CODE REVIEW

File: corehq/apps/ota/views.py [hunk 1]
```diff
@@ -74,9 +74,17 @@ def search(request, domain):
 @json_error
 @login_or_digest_or_basic_or_apikey()
 def claim(request, domain):
+    """
+    Allows a user to claim a case that they don't own.
+    """
+    # import ipdb; ipdb.set_trace()
```
Comment (proteusvacuum) [inline corehq/apps/ota/views.py]: `ipdb`

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 2]
```diff
@@ -88,6 +89,24 @@ def test_duplicate_claim(self):
         self.assertEqual(response.content, 'You have already claimed that case')
 
     @run_with_all_backends
+    def test_duplicate_user_claim(self):
```
Comment (proteusvacuum) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: nit: would be good to test the `get_first_claim` method directly too, even though this seems to serve that purpose.

File: corehq/apps/ota/views.py (hunk 1 above)
Comment (kaapstorm) [inline corehq/apps/ota/views.py]: :flushed: I used to have a pre-commit hook to catch that. I wonder what happened to it.

**Comment by kaapstorm** [comment]
Tests added.

**Comment by proteusvacuum** [comment]
Thanks for following up!

**Pull request merged**